File: AMF5/src/test/scala/scalaPlatform/ConfigurationOptionsTest.scala
package scalaPlatform

import amf.apicontract.client.scala.OASConfiguration
import amf.core.client.scala.config.{ParsingOptions, RenderOptions}
import amf.core.internal.remote.Mimes
import amf.core.internal.remote.Mimes.`application/ld+json`
import org.junit.runner.RunWith
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should
import org.scalatestplus.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class ConfigurationOptionsTest extends AsyncFlatSpec with should.Matchers {

  "Configuration" should "be configurable with options" in {
    val client = OASConfiguration.OAS20()
      .withParsingOptions(ParsingOptions().setMaxYamlReferences(20))
      .withRenderOptions(RenderOptions().withSourceMaps).baseUnitClient()
    client.parse("file://src/test/resources/examples/banking-api.json") map { result =>
      val jsonld = client.render(result.baseUnit, `application/ld+json`)
      jsonld should include("[(87,12)-(87,39)]")
    }
  }
}

File: AMF5/src/test/java/javaPlatform/ConfigurationOptionsTest.java
package javaPlatform;

import amf.apicontract.client.platform.AMFBaseUnitClient;
import amf.apicontract.client.platform.OASConfiguration;
import amf.core.client.platform.AMFResult;
import amf.core.client.platform.config.ParsingOptions;
import amf.core.client.platform.config.RenderOptions;
import amf.core.internal.remote.Mimes;
import org.junit.Test;

import java.io.IOException;
import java.util.concurrent.ExecutionException;

public class ConfigurationOptionsTest {

    @Test
    public void configurationWithOptions() throws ExecutionException, InterruptedException, IOException {
        AMFBaseUnitClient client = OASConfiguration.OAS20()
                .withParsingOptions(new ParsingOptions().setMaxYamlReferences(20))
                .withRenderOptions(new RenderOptions().withSourceMaps())
                .baseUnitClient();

        final AMFResult parseResult = client.parse("file://src/test/resources/examples/banking-api.json").get();
        final String result = client.render(parseResult.baseUnit(), "application/ld+json").trim();
        assert(result.contains("[(87,12)-(87,39)]"));
    }
}
